packages/go_router/lib/src/state.dart

```diff
@@ -252,3 +255,208 @@ class GoRouterStateRegistry extends ChangeNotifier {
     }
   }
 }
+
+/// The snapshot of the current state of a [StatefulShellRoute].
+///
+/// Note that this an immutable class, that represents the snapshot of the state
+/// of a StatefulShellRoute at a given point in time. Therefore, instances of
+/// this object should not be stored, but instead fetched fresh when needed,
+/// using the method [StatefulShellRoute.of].
+@immutable
+class StatefulShellRouteState {
```

Comment:
You may be right, if the intention of state.dart was only to host `GoRouterState`? Although I'm wondering if there is a better option than moving to stateful_navigation_shell.dart, since StatefulShellRouteState and StatefulShellBranchState are public API, and StatefulNavigationShell isn't. Also, stateful_navigation_shell.dart is already quite long, so perhaps a new file would be better, like `shell_state.dart` or something?